pkg/cloud/gcp.go

```diff
@@ -0,0 +1,39 @@
+package cloud
+
+import (
+	"strings"
+
+	"cloud.google.com/go/compute/metadata"
+	"github.com/pkg/errors"
+)
+
+// gcpHostnameMaxLength is the maximum number of character allowed for the instance's hostname in GCP
+const gcpHostnameMaxLength = 63
+
+// processGCPInstanceHostname truncates the given hostname if the length is longer 63 characters, otherwise returns
+// the original value
+func processGCPInstanceHostname(hostname string) string {
```

Comment:
Is there an existing function in MAO or MCO that you can leverage?

Reply:
I looked and couldn't find anything.